ghcide/src/Development/IDE/Spans/AtPoint.hs

```diff
@@ -205,6 +206,44 @@ gotoDefinition
 gotoDefinition withHieDb getHieFile ideOpts imports srcSpans pos
   = lift $ locationsAtPoint withHieDb getHieFile ideOpts imports pos srcSpans
 
+-- | Synopsis for the package at a given position.
+atPointPackage :: HieAstResult
+  -> HscEnv
+  -> Position
+  -> IO (Maybe (Maybe Range, [T.Text]))
+atPointPackage (HAR _ hf _ _ kind) env pos =
+  case uncons (pointCommand hf pos hoverInfo) of
```

Comment:
Yes you are right, ideally, I just want to prevent impurity in `atPioint`.

I've checked `HieDb` and found it doesn't have any clue about `import` statements, so I'm thinking if I can add an extra field in `HieDb` to prevent doing IO for every hover.